src/envoy/mixer/mixer_control.cc

```diff
@@ -56,11 +60,35 @@ void CreateEnvironment(Upstream::ClusterManager& cm,
 
 }  // namespace
 
+MixerControlBase::MixerControlBase(Event::Dispatcher& dispatcher,
+                                   const std::string& stats_prefix,
+                                   Stats::Scope& scope)
+    : stats_(stats_prefix, scope), dispatcher_(dispatcher) {}
+
+void MixerControlBase::SetUpStatsTimer() {
+  timer_ = dispatcher_.createTimer([this]() -> void { StatsUpdateCallback(); });
+  timer_->enableTimer(std::chrono::milliseconds(MixerStatsObject::kStatsUpdateIntervalInMs));
+}
+
+void MixerControlBase::StatsUpdateCallback() {
+  ::istio::mixer_client::Statistics new_stats;
+  stats_.GetStatistics(&new_stats);
+  stats_.CheckAndUpdateStats(new_stats);
+  timer_->enableTimer(std::chrono::milliseconds(MixerStatsObject::kStatsUpdateIntervalInMs));
```

Comment:
I don't think you need to call "enable" for each call.  Only need to enable it once.